crypto/ec/ec_lib.c

```diff
@@ -66,13 +66,13 @@ EC_GROUP *EC_GROUP_new(const EC_METHOD *meth)
 void EC_pre_comp_free(EC_GROUP *group)
 {
     switch (group->pre_comp_type) {
-    default:
+    case PCT_none:
         break;
-#ifdef ECP_NISTZ256_REFERENCE_IMPLEMENTATION
     case PCT_nistz256:
+#ifdef ECP_NISTZ256_REFERENCE_IMPLEMENTATION
```

Comment:
Following comment is not directly related to this request, but there seem to be gross misunderstanding regarding ECP_NISTZ256_REFERENCE_IMPLEMENTATION macro. It's *not* normally defined even if ecp_nistz256 module is compiled. For reference, macro is used when new ecp_nistz256 assembly module is being developed. One starts by configuring with -DECP_NISTZ256_REFERENCE_IMPLEMENTATION and implementing only functions performing simplest arithmetic operations on 256-bit vectors. Once they work one proceeds to implementation of higher-level functions performing *point* operations. Then it's time to remove ECP_NISTZ256_REFERENCE_IMPLEMENTATION and *never* define it again. Correct macro denoting presence of ecp_nistz256 module is ECP_NISTZ256_ASM.